### ISSUE TITLE
prSetEdited errors on Windows IDE when saving files

### ISSUE THREAD

**Issue by eleses** [supercollider/supercollider#4812]
## Environment

* SuperCollider version: 3.10.4
* Operating system: Windows 10
* Other details (Qt version, audio driver, etc.):

## Steps to reproduce

This may be quite hard to repro, but on Win 10, after hammering at various open documents for a while, I sometimes get this prSetEdited error (output below) when I hit Ctrl+S to save a document... Yes, I'm quite sure it's the IDE throwing this because it happened even when I have no code dealing with any GUI of my own in any open document, or in any document even open & closed since starting the IDE...

The actual saving of the file seems to work properly though, despite this error, so it's probably just "cosmetic". I haven't modified my IDE in any way, but I do use (i.e. keep open) the separate "Documents" doclet since the modified-needs-saving icon is no longer displayed on the name tabs in the newer versions of SC, but the icon is still displayed in that doclet next to the files that have been changed since their last save. Given the name of the method involved in this error thrown, it's probably related to that doclet...

Looking around at other bug reports I see in #3054 that the tabbed view and the Documents doclet use different icons. In my installation of SC on Win 10, I don't get *any* icon in the tabbed view, ever, only the Documents doclet displays those (blue ones). I'm not sure this a bug too or just a newer "feature". I assumed the latter since the IDE was changed to  a different internal editor since the days of SC 3.6... if I'm not mistaken about that.

## Expected vs. actual behavior

```
ERROR: Message 'prSetEdited' not understood.
RECEIVER:
   nil
ARGS:
   Integer 0
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Object:doesNotUnderstand
		arg this = nil
		arg selector = 'prSetEdited'
		arg args = [*1]
	Process:interpretCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Message 'prSetEdited' not understood.
RECEIVER: nil
```

**Comment by totalgee** [comment]
I also get this all the time (on Windows 10), especially now with 3.11.0 it seems even more frequent, but I also had it often with 3.10.3 and 3.10.4 for sure.

It happens, e.g. if I have a class file (.sc) open, save it, then press Ctrl-Shift-L to recompile the class library, and then make an edit of the open (saved) file (e.g. press 'space' or hit enter) somewhere in the document. It is really annoying (seems more likely to happen than before, I think).

I also get this other message sometimes, which I think is related to the above problem...(?)

```txt
ERROR: Message 'didBecomeKey' not understood.
RECEIVER:
   nil
ARGS:
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Object:doesNotUnderstand
		arg this = nil
		arg selector = 'didBecomeKey'
		arg args = [*0]
	Meta_Document:prCurrent_
		arg this = <instance of Meta_Document>
		arg newCurrent = nil
	Meta_Document:setActiveDocByQUuid
		arg this = <instance of Meta_Document>
		arg quuid = '{ce76b6c0-4c11-4791-b459-4b92a99c8045}'
		var newCurrent = nil
	< closed FunctionDef >  (no arguments or variables)
	Process:interpretCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Message 'didBecomeKey' not understood.
RECEIVER: nil
```

**Comment by jrsurge** [comment]
This seems to be a duplicate of #4413?

**Comment by brianlheim** [comment]
agreed, these all seem to be the same underlying issue. closing and moving some of these comments there

**Comment by brianlheim** [comment]
ah actually i'm not sure if the original ticket here is the same issue, but @totalgee's definitely is. more investigation needed

**Comment by eleses** [comment]
I'm hardly an expert on the IDE, but since the document does save (unlike the other error where it doesn't close), it ineed might not be the exact same underlying reason.

**Comment by eleses** [comment]
Also interestingly perhaps, it bears some similarity to a very old bug report https://github.com/supercollider/supercollider/issues/1220

**Comment by Spacechild1** [comment]
I just wanted to chime in and say that I also get those errors all the time on Windows (using SC 3.11.1).

**Comment by bembidiona** [comment]
yep, still happening here too in SC v3.11.1 x64 WIN 10

**Comment by brianlheim** [comment]
we should probably make it a priority to fix this for the 3.12 release, if it's not too tricky

**Comment by sternsc** [comment]
I'm still getting this in SC v3.12.2 (x64 Win 10), on the first char to be typed into an IDE doc. Same error message as reported above on March 15, 2020.

**Comment by dyfer** [comment]
@sternsc does the problem still occur in 3.13.0-rc?

**Comment by sternsc** [comment]
Yes, it does. 
Another error that seems to be related is this; it occurs when switching document tabs in the IDE editor after having recompiled a help file, sometimes also after a class library recompile. 

ERROR: Message 'didBecomeKey' not understood.
RECEIVER:
   nil
ARGS:
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Object:doesNotUnderstand
		arg this = nil
		arg selector = 'didBecomeKey'
		arg args = [*0]
	Meta_Document:prCurrent_
		arg this = <instance of Meta_Document>
		arg newCurrent = nil
	Meta_Document:setActiveDocByQUuid
		arg this = <instance of Meta_Document>
		arg quuid = '{a28eb73c-86d0-4948-849a-17f87312d87d}'
		var newCurrent = nil
	< closed FunctionDef >  (no arguments or variables)
	Process:interpretCmdLine
		arg this = <instance of Main>
^^ ERROR: Message 'didBecomeKey' not understood.
RECEIVER: nil

**Comment by JordanHendersonMusic** [comment]
I've closed some duplicate, this issue often arises in conjunction with #4413
